packages/design-system/src/components/N8nBreadcrumbs/Breadcrumbs.vue

```diff
@@ -0,0 +1,319 @@
+<script lang="ts" setup>
+import { computed, ref, watch } from 'vue';
+
+import type { UserAction } from 'n8n-design-system/types';
+
+import N8nLoading from '../N8nLoading';
+
+export type PathItem = {
+	id: string;
+	label: string;
+	href?: string;
+};
+
+type Props = {
+	items: PathItem[];
+	hiddenItems?: PathItem[] | Promise<PathItem[]>;
+	theme?: 'small' | 'medium';
+	showBorder?: boolean;
+	loadingSkeletonRows?: number;
+	separator?: string;
+	highlightLastItem?: boolean;
+};
+
+defineOptions({ name: 'N8nBreadcrumbs' });
+
+const emit = defineEmits<{
+	tooltipOpened: [];
+	tooltipClosed: [];
+	hiddenItemsLoadingError: [error: unknown];
+	itemSelected: [item: PathItem];
+}>();
+
+const props = withDefaults(defineProps<Props>(), {
+	hiddenItems: () => [],
+	theme: 'medium',
+	showBorder: false,
+	loadingSkeletonRows: 3,
+	separator: '/',
+	highlightLastItem: true,
+});
+
+const loadedHiddenItems = ref<PathItem[]>([]);
+const isLoadingHiddenItems = ref(false);
+const currentPromise = ref<Promise<PathItem[]> | null>(null);
+
+const hasHiddenItems = computed(() => {
+	return Array.isArray(props.hiddenItems)
+		? props.hiddenItems.length > 0
+		: props.hiddenItems instanceof Promise;
+});
+
+const hiddenItemActions = computed((): UserAction[] => {
+	return loadedHiddenItems.value.map((item) => ({
+		value: item.id,
+		label: item.label,
+		disabled: false,
+	}));
+});
+
+const getHiddenItems = async () => {
+	// If we already have items loaded and the source hasn't changed, use cache
+	if (loadedHiddenItems.value.length > 0 && props.hiddenItems === currentPromise.value) {
+		return;
+	}
+
+	// Handle synchronous array
+	if (Array.isArray(props.hiddenItems)) {
+		loadedHiddenItems.value = props.hiddenItems;
+		return;
+	}
+
+	isLoadingHiddenItems.value = true;
+	try {
+		// Store the current promise for cache comparison
+		currentPromise.value = props.hiddenItems;
+		const items = await props.hiddenItems;
+		loadedHiddenItems.value = items;
+	} catch (error) {
+		loadedHiddenItems.value = [];
+		emit('hiddenItemsLoadingError', error);
+	} finally {
+		isLoadingHiddenItems.value = false;
+	}
+};
+
+watch(
+	async () => await props.hiddenItems,
```

Comment:
Will this trigger the evaluation immediately? An important conceptual part of providing an async promise as a parameter is that it will remain unused if it wasn't needed.

This satisfies the linter, if that's the concern:
```suggestion
	(): PathItem[] | Promise<PathItem[]> => props.hiddenItems,
```
(probably need to get rid of `deep` at that point though)